pkg/pillar/cmd/loguploader/loguploader.go

```diff
@@ -722,22 +722,14 @@ func doFetchSend(ctx *loguploaderContext, zipDir string, iter *int) int {
 }
 
 func buildAppUUIDMap(fName string) {
-	var appUUID string
-	if strings.HasPrefix(fName, types.AppPrefix) && strings.HasSuffix(fName, ".gz") {
-		fStr1 := strings.TrimPrefix(fName, types.AppPrefix)
-		fStr := strings.Split(fStr1, types.AppSuffix)
-		if len(fStr) != 2 {
-			err := fmt.Errorf("app split is not 2")
-			log.Error(err)
-			return
-		}
-		appUUID = fStr[0]
+	appUUID, err := types.GetUUIDFromFileName(fName)
+	if err != nil {
+		log.Errorf("buildAppUUIDMap: cannot parse app file name %s: %v", fName, err)
```

Comment:
Could this generate/log an error when it is run on a device log filename?

Reply:
it would, because a device log filename doesn't contain a UUID. however this function is only run on app log files (see https://github.com/lf-edge/eve/blob/b703eb27e9f11419d768e492e7979f745df15e1d/pkg/pillar/cmd/loguploader/loguploader.go#L653)